libraries/rush-lib/src/logic/operations/ShardedPhaseOperationPlugin.ts

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type { IPhase } from '../../api/CommandLineConfiguration';
+import type { RushConfigurationProject } from '../../api/RushConfigurationProject';
+import type {
+  ICreateOperationsContext,
+  IPhasedCommandPlugin,
+  PhasedCommandHooks
+} from '../../pluginFramework/PhasedCommandHooks';
+import { Operation } from './Operation';
+import type { IOperationSettings, RushProjectConfiguration } from '../../api/RushProjectConfiguration';
+import { ShellOperationRunner } from './ShellOperationRunner';
+import {
+  getCustomParameterValuesByPhase,
+  getDisplayName,
+  getScriptToRun
+} from './ShellOperationRunnerPlugin';
+import { NullOperationRunner } from './NullOperationRunner';
+import { OperationStatus } from './OperationStatus';
+
+export const PLUGIN_NAME: 'ShardedPhasedOperationPlugin' = 'ShardedPhasedOperationPlugin';
+
+/**
+ * Phased command that shards a phase into multiple operations.
+ */
+export class ShardedPhasedOperationPlugin implements IPhasedCommandPlugin {
+  public apply(hooks: PhasedCommandHooks): void {
+    hooks.createOperations.tap(PLUGIN_NAME, spliceShards);
+  }
+}
+
+function spliceShards(existingOperations: Set<Operation>, context: ICreateOperationsContext): Set<Operation> {
+  const { projectConfigurations, rushConfiguration } = context;
+
+  const shardableOperations: Set<Operation> = new Set<Operation>();
+
+  for (const operation of existingOperations) {
+    const { associatedPhase: phase, associatedProject: project } = operation;
+    const operationSettings: IOperationSettings | undefined = getOperationSettings(phase, project);
+    if (operationSettings?.sharding) {
+      shardableOperations.add(operation);
+    }
+  }
+  const getCustomParameterValuesForPhase: (phase: IPhase) => ReadonlyArray<string> =
+    getCustomParameterValuesByPhase();
+
+  for (const operation of shardableOperations) {
+    const { associatedPhase: phase, associatedProject: project } = operation;
+    const operationSettings: IOperationSettings | undefined = getOperationSettings(phase, project);
+    if (phase && project && operationSettings?.sharding && operationSettings.sharding.count > 1) {
```

Comment:
It isn't safe to assume that just because the operation is configured to have exactly 1 shard that it doesn't need collation. It could have a build script for which the sharding arguments are mandatory.

Reply:
Isn't sharding of 1 and then collating the same as having 2 phases where one depends on the other? I removed the check already, but this feels like it this would be an incorrect usage of shards. You could just call the phases with `--shard 1/1` and `--output-directory my-output-directory` manually.